airbyte-cdk/python/connector_builder/connector_builder_handler.py

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from datetime import datetime
+from typing import Union
+
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+
+
+def list_streams() -> AirbyteRecordMessage:
+    raise NotImplementedError
+
+
+def stream_read() -> AirbyteRecordMessage:
+    raise NotImplementedError
+
+
+def resolve_manifest(source) -> Union[AirbyteMessage, AirbyteRecordMessage]:
```

Comment:
It feels a little inconsistent that we return either an AirbyteMessage or an AirbyteRecordMessage. The "parent" AirbyteMessage object has fields for each of the relevant message unless the protocol has changed. Shouldn't we be returning an AirbyteMessage w/ the `record` or `trace` field set and the relevant `type`. Or returning a `Union[AirbyteRecordMessage, AirbyteTraceMessage]`?

Reply:
Thanks for pointing that out, I think always returning `AirbyteMessage` makes sense.

Reply:
(updated)